Add "Trusted-Coding Static Analysis" package

innovation-madnext opened the pull request:
<!--
Your pull request will be reviewed automatically and by a human.

The manual review may take several days or weeks,
depending on the reviewer's availability and workload.
If you haven't received a comment on your pull request
and it wasn't merged either,
it just hasn't been reviewed yet.

---

Please ensure the automated reviews pass.
Follow the instructions provided, if necessary.
You can speed up the process
by [running some tests locally](https://packagecontrol.io/docs/submitting_a_package#Step_7).

You can request a review from @packagecontrol-bot
to manually trigger an automated review
if you don't need to push a new commit.
Do **NOT** open a new pull request!

In general, make sure you:

 1. Used `"tags": true` and not `"branch": "master"`
    (versioning docs: <https://packagecontrol.io/docs/submitting_a_package#Step_4>)
 2. Added a README to your repository so that users (and reviewers) 
    can understand what your package provides.

You should proceed with a short description of what the package does
and, in case one or multiple similar package already exists,
why you believe it is different and needed
below this line. -->

braver commented:
Please, elaborate: what kind of package and why should it be added?

innovation-madnext commented:
The core of the package is that it can analyze statically PHP and Python to find security flaws. When scan has finished, it show all the flaws linked to OWASP and/or CWE weakness, so, the user can see a brief description and related information to those weaknesses.

braver commented on repository/t.json:
```diff
@@ -3246,6 +3246,17 @@
+				{
+					"sublime_text": "*",
+					"tags": true,
+					"platforms": "*"
```
If platforms is "*" please leave it out.

innovation-madnext replied:
Ok! Removed in the new commit

braver commented on repository/t.json:
```diff
@@ -3246,6 +3246,17 @@
 		},
+		{
+			"name": "Trusted Coding",
+			"details": "https://bitbucket.org/madnext/trustedcodingsublime",
```
You might want to add some labels for categorization of your package.

innovation-madnext replied:
I just added some labels. Thanks! :)

braver reviewed (changes requested):
Your package seems to have neither a description nor a README. Without those people will not be able to discover your package and know what it's for, let alone know how to use it.

You ship with keybindings. This is not recommended because you will inevitably conflict with other packages. Also, it does not sound like this is something people need to have at their finger tips at all times. Besides, keybindings are not discoverable. Which is why we have a command palette. Please at the very least add your commands there.
Something similar goes for the context menu. Unless you really need to be in there, you should try to stay out of it: it's a very limited space and everyone wants their package in that menu. You already have entries in the Tools menu, which is great, and if you're in the command palette, usually that's more than enough.

It feels like you're either providing settings via that menu, or actually providing a wizard. If you're exposing setting that way, please consider making a settings file, or supporting project settings if you expect people to want different settings for different cases.

rchl commented:
- Your package is configured to connect to a local API server - https://bitbucket.org/madnext/trustedcodingsublime/src/d516341c2f93f09c427c2d8b2c06dc448391ad62/TrustedCoding.py#lines-204:205
 - If your package is sending the whole checked code into some remote API server you should make it very clear in the readme. And also have a license that is being specific on how you handle the data (I'm assuming it's your server)

innovation-madnext commented:
> 
> 
> Your package seems to have neither a description nor a README. Without those people will not be able to discover your package and know what it's for, let alone know how to use it.
> 
> You ship with keybindings. This is not recommended because you will inevitably conflict with other packages. Also, it does not sound like this is something people need to have at their finger tips at all times. Besides, keybindings are not discoverable. Which is why we have a command palette. Please at the very least add your commands there.
> Something similar goes for the context menu. Unless you really need to be in there, you should try to stay out of it: it's a very limited space and everyone wants their package in that menu. You already have entries in the Tools menu, which is great, and if you're in the command palette, usually that's more than enough.
> 
> It feels like you're either providing settings via that menu, or actually providing a wizard. If you're exposing setting that way, please consider making a settings file, or supporting project settings if you expect people to want different settings for different cases.

Removed context menu and keybindings. There is only two ways to execute the package options now, command palette and main menu.
Also added a README in the package repository with description, usage,... (PENDING LEGAL)

innovation-madnext commented:
> 
> 
>     * Your package is configured to connect to a local API server - https://bitbucket.org/madnext/trustedcodingsublime/src/d516341c2f93f09c427c2d8b2c06dc448391ad62/TrustedCoding.py#lines-204:205
> 
>     * If your package is sending the whole checked code into some remote API server you should make it very clear in the readme. And also have a license that is being specific on how you handle the data (I'm assuming it's your server)

Fixed: Endpoint removed, now pointing to prod.
Pending: I'm waiting for the legal department. At the moment, it has only been indicated in the README that the code will be sent to our backend, where it will be analyzed and processed. So, if You want, this PR can be hold on until the legal part is ready...

braver commented:
Looks good to me. I don't have a project on hand to test it with, but it looks sane enough and has nothing that should harm performance or anything. The information in the README is clear enough, thanks for that. A license isn't a requirement, but a recommendation. I also don't necessarily need to wait for your legal department to clear it. So, as far as I can see it's good to go. Let me know if you're ready to merge this.

braver commented:
>  Let me know if you're ready to merge this.

So, yeah, give me a ping when you're ready...

innovation-madnext commented:
Sorry to keep you waiting! Legal department delayed the response a lot... The "Terms and Conditions" are ready in the repo's README :) I would really appreciate it if you accept the PR :) Thanks!
